apps/dashboard/src/components/layouts/layout-control-input.tsx

```diff
@@ -0,0 +1,131 @@
+import { EditorView } from '@uiw/react-codemirror';
+import { cva } from 'class-variance-authority';
+import { useMemo, useRef } from 'react';
+import { CompletionRange, VariableEditor } from '@/components/primitives/variable-editor';
+import { useEditorTranslationOverlay } from '@/hooks/use-editor-translation-overlay';
+import { LocalizationResourceEnum } from '@/types/translations';
+import { IsAllowedVariable, LiquidVariable } from '@/utils/parseStepVariables';
+import { cn } from '@/utils/ui';
+import { EditTranslationPopover } from '../workflow-editor/steps/email/translations/edit-translation-popover/edit-translation-popover';
+import { useLayoutEditor } from './layout-editor-provider';
+
+const variants = cva('relative w-full', {
+  variants: {
+    size: {
+      md: 'p-2.5',
+      sm: 'p-2',
+      '2xs': 'px-2 py-1.5',
+      '3xs': 'px-1.5 py-1 text-xs',
+    },
+  },
+  defaultVariants: {
+    size: 'sm',
+  },
+});
+
+type LayoutControlInputProps = {
+  className?: string;
+  value: string;
+  onChange: (value: string) => void;
+  onBlur?: () => void;
+  variables: LiquidVariable[];
+  isAllowedVariable: IsAllowedVariable;
+  placeholder?: string;
+  autoFocus?: boolean;
+  size?: 'md' | 'sm' | '2xs' | '3xs';
+  id?: string;
+  multiline?: boolean;
+  indentWithTab?: boolean;
+  enableTranslations?: boolean;
+  disabled?: boolean;
+};
+
+export function LayoutControlInput({
```

Comment:
This is the control input that has the layout context inside. It is used in the translation popover as the value input where the user can type variables.